### PR TITLE
Adding phase() method to BaseManager

### PR DESCRIPTION
**Pull request by corey-nm**
Adds a `phase()` method to manager intended to be used when saving checkpoints. See docstring of method for intended use.

# Test plan

added unit tests

### CODE REVIEW

**Comment by dbogunowicz** [comment]
@corey-nm beautiful PR

**Review by dbogunowicz — DISMISSED** [review]

File: src/sparseml/optim/manager.py [hunk 1]
```diff
@@ -561,6 +561,67 @@ def qat_active(self, epoch: float) -> bool:
...
+        pruners: List[BaseModifier] = self.pruning_modifiers
+        quantizers: List[BaseModifier] = self.quantization_modifiers
+
+        if len(pruners) == 0:
+            pruned = False
+            pruning_in_progress = False
+        else:
+            pruning_start = min(mod.start_epoch for mod in pruners)
+            pruning_end = max(mod.end_epoch for mod in pruners)
+            pruning_in_progress = pruning_start <= epoch <= pruning_end
```
Comment (corey-nm) [inline src/sparseml/optim/manager.py:598]: Should this be:
```suggestion
            pruning_in_progress = pruning_start <= epoch < pruning_end
```
?

Comment (bfineran) [reply src/sparseml/optim/manager.py:598]: has `epoch` already been applied? pruning is "complete" after the pruning modifier performs its `update` when `epoch==end_epoch`

Comment (corey-nm) [reply src/sparseml/optim/manager.py:598]: The intention is yes - although I guess that depends on when this is called currently.

File: src/sparseml/optim/manager.py [hunk 2]
```diff
@@ -561,6 +561,67 @@ def qat_active(self, epoch: float) -> bool:
...
+            pruned = epoch > pruning_end
+
+        if len(quantizers) == 0:
+            quantized = False
+            quantization_in_progress = False
+        else:
+            first_quant_epoch = min(mod.start_epoch for mod in quantizers)
+            last_quant_epoch = max(mod.start_epoch for mod in quantizers)
+            quantization_in_progress = first_quant_epoch <= epoch <= last_quant_epoch
+            quantized = epoch > last_quant_epoch
```
[the remark this answers is not in the document]
Comment (bfineran) [reply src/sparseml/optim/manager.py:608]: same as above, depends on application. also `quantization_in_progress` may be a bit of a misnomer - in general best practice is to use 1 quantization modifier (programming for multiple is the goal to add some robustness), however `quantization` itself, specifically QAT is always happening. the `in_progress` portion is just when the graph is still not updated to its final quantized state

**Review by KSGulin — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

**Pull request merged**